### PR TITLE
Supervised learning

### PR DESCRIPTION
**Pull request by benediamond**
Implement supervised learning features, following @Zeta36 and @Akababa.

### CODE REVIEW

**Comment by benediamond** [comment]
@Zeta36, @Akababa, @bame55, though I've attempted to mimick the DeepMind network architecture, I'm having difficulty achieving convergence, stuck around
`loss: 6.3532 - policy_out_loss: 4.9872 - value_out_loss: 0.6904`
after a few hours of training on KingBase games with no sign of improvement.

Though I hate to ask for general help here, I'm just at a loss for how to get this to converge. I've tried Xavier-initializing the weights of the convolutional layers, and it didn't seem to help. i.e.,
```python
x = Conv2D(filters=2, kernel_size=1, data_format="channels_first", kernel_initializer='glorot_normal', bias_initializer='zeros', kernel_regularizer=l2(mc.l2_reg))(res_out)
```
Is there any secret you can think of, that might get this going properly? :( I can't find any bugs in the feature representation. I'd really appreciate your input.

**Comment by benediamond** [comment]
I will continue to let this run overnight but it doesn't look good so far.

**Comment by Akababa** [comment]
:( Sorry to hear that, but in my opinion at the original 1:1 weights that's not so bad.
I made some notes on overfitting [here](https://github.com/Akababa/chess-alpha-zero/issues/2), maybe you could check it out and add your thoughts.

**Comment by Zeta36** [comment]
@benediamond, I faced the same problem when I tried to converge the model I created mimicking the DeepMind planes feed input. It's more, my model also got stuck around a loss of 6 like yours. That's why I restored master branch again to the 2 planes version erasing my changes in that sense.

**Comment by Zeta36** [comment]
I wonder if you @Akababa were able to converge your model once you introduce more planes of input information.

**Comment by benediamond** [comment]
@Zeta36 @Akababa This is frustrating. My best guess is either some mistake in implementation...?! or, with a different amount of input planes, the **NN parameters will have to be changed somehow**. This could include...

1) **L2 weight regularization.** I've wondered if our regularization constant (currently 1e-4) is too high. Thus, we're preventing overfitting "too much", and restricting convergence.
2) **The optimizer.** The momentum of 0.9 and the learning rate annealing schedule are given by DeepMind. But perhaps we should consider using Nesterov momentum, or even another optimizer, such as `Adagrad` or `Adam` instead of `SGD`.
3) **Weight initialization** for the convolutional kernels, etc., e.g. Xavier initialization (though I tried this and it didn't look good...).

I'll experiment with these; I'm happy to hear further thoughts.
